sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/EntityDescriptorFactory.scala

```diff
@@ -16,23 +16,55 @@
 
 package kalix.springsdk.impl
 
+import kalix.EntityMethodOptions.Generator
 import kalix.springsdk.annotations.EntityKey
+import kalix.springsdk.annotations.GenerateEntityKey
 import kalix.springsdk.impl.ComponentDescriptorFactory.buildJWTOptions
 import kalix.springsdk.impl.reflection.KalixMethod
 import kalix.springsdk.impl.reflection.NameGenerator
 import kalix.springsdk.impl.reflection.RestServiceIntrospector
+import kalix.springsdk.impl.reflection.ServiceIntrospectionException
 
 private[impl] object EntityDescriptorFactory extends ComponentDescriptorFactory {
 
   override def buildDescriptorFor(
       component: Class[_],
       messageCodec: SpringSdkMessageCodec,
       nameGenerator: NameGenerator): ComponentDescriptor = {
-    val entityKeys = component.getAnnotation(classOf[EntityKey]).value()
+
+    val entityKeysOnType = {
+      val anno = component.getAnnotation(classOf[EntityKey])
+      if (anno != null) anno.value()
+      else Array.empty[String]
+    }
 
     val kalixMethods =
       RestServiceIntrospector.inspectService(component).methods.map { restMethod =>
```

Comment:
If the user adds `EntityKey` or `GeneratedEntityKey` to more than one method the last method wins?
If this is the case, Would be better to add validation so only one method can have one annotation or the other?

Reply:
oh sorry. I misunderstood. Never mind. There will be one annotation on the type or one per method with GET/POST. Although now this makes me think. Would make sense to add validation that there is at least one instance of 'EntityKey' or 'GeneratedEntityKey' that the rest method can use?